langchain/chains/serverless/memory.py

```diff
@@ -0,0 +1,94 @@
+import boto3
+from langchain.chains.base import Memory
+from pydantic import BaseModel
+from typing import Any, Dict, List, Optional
+from datetime import datetime
+
+now = datetime.now()
+
+def _get_prompt_input_key(inputs: Dict[str, Any], memory_variables: List[str]) -> str:
+    # "stop" is a special key that can be passed as input but is not used to
+    # format the prompt.
+    prompt_input_keys = list(set(inputs).difference(memory_variables + ["stop"]))
+    if len(prompt_input_keys) != 1:
+        raise ValueError(f"One input key expected got {prompt_input_keys}")
+    return prompt_input_keys[0]
+
+class ConversationBufferDynamoDBMemory(Memory, BaseModel):
+    """Buffer for storing conversation memory in DynamoDB."""
+
+    human_prefix: str = "Human"
+    ai_prefix: str = "AI"
+    """Prefix to use for AI generated responses."""
+    output_key: Optional[str] = None
+    input_key: Optional[str] = None
+    memory_key: str = "chat_history"
+    table_name: str = "ConversationHistory"
+    buffer: str = ""
+    parition_key: str = "id"
+    id: str
```

Comment:
Considering that this will correspond with a conversation session, does it make sense to call this `session_id`?